### PR TITLE
campaigns: remove licence checks

### PR DESCRIPTION
**Pull request by LawnGnome**
In the future, we should be able to use the revert of this PR as a starting point for reintroducing licence checks.

Related to #15779, although there'll be more PRs to address other aspects of this.

### CODE REVIEW

**Comment by unknwon** [comment]
What is the context of this change?

**Comment by LawnGnome** [comment]
The context for this is #15779. (Sorry, I had this verbally on Friday, but @rrhyne was still working out the details of what we were going to do for this release.)

**Review by eseliger — APPROVED** [review]
This is basically a revert of the PR that introduced it, right?
LGTM regardless

**Comment by LawnGnome** [comment]
> This is basically a revert of the PR that introduced it, right?

More or less, yes. (It wasn't _quite_ that neat in practice, since we'd made some changes since to the tests, but it's pretty close.)

File: enterprise/internal/campaigns/resolvers/resolver.go [hunk 1]
```diff
@@ -66,24 +64,6 @@ func campaignsCreateAccess(ctx context.Context) error {
 	return nil
 }
 
-// checkLicense returns a user-facing error if the campaigns feature is not purchased
-// with the current license or any error occurred while validating the license.
-func checkLicense() error {
-	if !licensing.EnforceTiers {
-		return nil
-	}
```
Comment (unknwon) [inline enterprise/internal/campaigns/resolvers/resolver.go:74]: To minimize the work of this PR and the future revert, I would suggest just add a direct return here with a note like "We're not enforcing Campaigns with a trail message until further decision."

Comment (mrnugget) [reply enterprise/internal/campaigns/resolvers/resolver.go:74]: We have git and can always revert this commit :) And I'm reasonably sure that the license check in future will be a bit more complex (maybe it won't be here in the resolver, but rather in the reconciler/another mutation, something which we talked about)

**Review by mrnugget — APPROVED** [review]

**Pull request merged**